pkg/processor/build/builder.go

```diff
@@ -924,10 +924,20 @@ func (b *Builder) createTempDir() error {
 
 	// either use injected temporary dir or generate a new one
 	if b.options.FunctionConfig.Spec.Build.TempDir != "" {
-		b.tempDir = b.options.FunctionConfig.Spec.Build.TempDir
 
-		err = os.MkdirAll(b.tempDir, 0744)
+		// Validate the user-provided temporary directory to ensure it does not contain directory traversal sequences
+		if strings.Contains(b.options.FunctionConfig.Spec.Build.TempDir, "..") {
+			return errors.New("Invalid temporary directory path: contains '..'")
+		}
```

Comment:
Is it really an issue?
What about a case where a user has this directory structure:
```
- project
--- temp
--- functions
```
then running from the functions dir, pointing to the temp dir with `../temp`.

I suppose we can limit the usage and force full paths, but I'm not sure it is necessary
```

Reply:
@TomerShor then we will need to have some resolving mechanism which would return the actual directory and then to check whether this directory is in `/tmp`. Taking into account that this parameter is not smth that users use a lot, not sure if we need to put an effort there